source/ast/symbols/MemberSymbols.cpp

```diff
@@ -2144,6 +2159,21 @@ std::span<const Expression* const> NetAliasSymbol::getNetReferences() const {
 
         netRef.visit(visitor);
         buffer.push_back(&netRef);
+
+        if (exprSyntax != *nets.begin()) {
```

Comment:
This logic seems confusing to me; you really just need to check whether the same symbol appears in more than one of the net alias expressions right? The `aliasedSyms` map in the NetAliasVisitor is already tracking that; instead of insert_or_assign up above you can just check whether the insertion fails because the symbol already exists, in which case you can issue the error right there. Then you don't need any of this special casing for it here and the separate map tracking the same thing.

Reply:
solution was reimplemented by focusing on merge net alias expressions 

`aliasedSyms` map logic was reduced from `NetAliasVisitor`

Reply:
My solution no longer ignores the leftoutermost net alias expression

Reply:
To relate net alias expressions to existing alias sybmols, I used a comparison of their syntaxes

it's not a best solution but in this case it is possible, since in aliases all symbols are global nets from the point of view of the module, in the future we need to come up with a better comparator

Reply:
> I'm not following why the first alias expression should be exempt from the duplicate checking. There isn't anything special about the first alias vs any others; you can reverse the order without changing the behavior.
> 
> I do think we need more info included in the addDriver call though, since it's only an error if the overlap occurs with the same target net? Like you can alias the same range of bits from one net to multiple different nets, it's only error if it's the same nets each time. This example from the LRM I think won't pass with the current code?
> 
> module overlap(inout wire [15:0] bus16, inout wire [11:0] low12, high12);
> alias bus16[11:0] = low12;
> alias bus16[15:4] = high12;
> endmodule
> 
> module overlap(inout wire [15:0] bus16, inout wire [11:0] low12, high12);
> alias bus16 = {high12, low12[3:0]};
> alias high12[7:0] = low12[11:4];
> endmodule
> Or maybe it will, but not if you reverse the ordering in the alias statements.

Provided example works correctly regardless of the alias permutations.